schemas/context/segmentmembership.schema.json

```diff
@@ -29,6 +29,83 @@
           "format": "date-time",
           "description": "The timestamp when the assertion of segment membership was made."
         },
+        "xdm:payload" : {
+          "title" : "Payload",
+          "description": "Values that are directly related with the segment realization. This payload exists with the same validUntil as the segment realization.",
+          "oneOf": [
+            {
+              "type": "object",
+              "properties": {
+                "xdm:payloadNumberValue": {
+                  "title": "Value",
+                  "type": "number",
+                  "description": "The number"
+                },
+                "xdm:payloadType": {
+                  "title": "Payload Type",
+                  "type": "string",
+                  "const": "number",
+                  "description": "The type of payload."
+                }
+              },
+              "required": ["xdm:payloadNumberValue", "xdm:payloadType"]
+            },
+            {
+              "type": "object",
+              "properties": {
+                "xdm:payloadPropensityValue": {
+                  "title": "Value",
+                  "type": "number",
+                  "description": "The propensity",
+                  "minimum": 0,
+                  "maximum": 1
+                },
+                "xdm:payloadType": {
+                  "title": "Payload Type",
+                  "type": "string",
+                  "const": "propensity",
+                  "description": "The type of payload."
+                }
+              },
+              "required": ["xdm:payloadPropensityValue", "xdm:payloadType"]
+            },
+            {
+              "type": "object",
+              "properties": {
+                "xdm:payloadBooleanValue": {
+                  "title": "Value",
+                  "type": "boolean",
+                  "description": "The boolean value"
+                },
+                "xdm:payloadType": {
+                  "title": "Payload Type",
+                  "type": "string",
+                  "const": "boolean",
+                  "description": "The type of payload."
+                }
+              },
+              "required": ["xdm:payloadBooleanValue", "xdm:payloadType"]
+            },
+            {
+              "type": "object",
+              "properties": {
+                "xdm:payloadStringValue": {
+                  "title": "Value",
+                  "type": "string",
+                  "description": "The string value"
+                },
+                "xdm:payloadType": {
+                  "title": "Payload Type",
+                  "type": "string",
+                  "const": "string",
+                  "description": "The type of payload."
+                }
+              },
+              "required": ["xdm:payloadStringValue", "xdm:payloadType"]
+            }
+
+          ]
+        },
```

Comment:
I don't see any metadata related to the nature of payload itself. The metadata is related to the payload value. Just by looking at data, I won't be able to find out what it represents.

Reply:
I think this belongs in the segment definition (which we haven't defined an xdm about yet). I think you'll have to join between segment definition and segment membership. Does that fit with how you see it, @rkpandey ?